File: nodejs/src/decorator/BobaPearlDecorator.ts
import BeverageDecorator from "./BeverageDecorator";

class BobaPearlDecorator extends BeverageDecorator {
    getDescription(): string {
        return `${this.beverage.getDescription()} + Trân châu`;
    }
    getCost(): number {
        return this.beverage.getCost() + 12000;
    }
}

export default BobaPearlDecorator;

File: spring/demo/src/main/java/com/example/demo/decorator/BobaPearlDecorator.java
package com.example.demo.decorator;

public class BobaPearlDecorator extends BeverageDecorator {
    public BobaPearlDecorator(Beverage beverage) {
        super(beverage);
    }

    @Override
    public String getDescription() {
        return beverage.getDescription() + " + Trân châu";
    }

    @Override
    public int getCost() {
        return beverage.getCost() + 12000;
    }
}